dlhub_sdk/utils/publish.py

```diff
@@ -0,0 +1,244 @@
+        serv_location = f'https://petreldata.net/dlhub/detail/{serv_location}'
+        visible_to = ['public']
+
+        dataset_identifier = client.create_identifier(
+            namespace=identifiers_namespace,
+            location=[serv_location],
+            metadata={
+                'uuid': task['dlhub']['id'],
+                'shorthand_name': task['dlhub']['shorthand_name']
+            },
+            landing_page=serv_location,
+            visible_to=visible_to)
+
+        res = dataset_identifier.data
+        identifier = res['identifier']
+        logger.debug(f"Created identifier: {identifier}")
+    except Exception as e:
+        logger.error(e)
+    return identifier
+
+
+def register_funcx(task, container_uuid, funcx_client):
+    """Register the function and the container with funcX.
+
+    Parameters
+    ----------
+    task : dict
+        A dict of the task to publish
+    container_uuid:  str
+    funcx_client:  the funcx client to register with
+
+    Returns
+    -------
+    str
+        The funcX function id
+    """
+
+    fxc = funcx_client
+
+    description = f"A container for the DLHub model {task['dlhub']['shorthand_name']}"
+    try:
+        description = task['datacite']['descriptions'][0]['description']
+    except:
+        # It doesn't have a simple description
+        pass
+    # I believe that the Container Service registers the container w/ funcx
+    # Register the container with funcX
+    # container_id = fxc.register_container(task['dlhub']['ecr_uri'], 'docker', name=task['dlhub']['shorthand_name'],
+    #                                      description=description)
+
+    # Register a function
+    funcx_id = fxc.register_function(dlhub_run, function_name=task['dlhub']['name'],
+                                     container_uuid=container_uuid, description=description, public=True)
+
+    # Whitelist the function on DLHub's endpoint
+    #endpoint_uuid = '86a47061-f3d9-44f0-90dc-56ddc642c000'
+    endpoint_uuid = '2238617a-8756-4030-a8ab-44ffb1446092'
+
+    res = fxc.add_to_whitelist(endpoint_uuid, [funcx_id])
+    print(res)
+    return funcx_id
+
+
+def dlhub_run(event):
+    """Invoke the DLHub servable"""
+    import json
+    import time
+    import os
+
+    from os.path import expanduser
+    path = expanduser("~")
+    os.chdir(path)
+
+    # Check to see if event is from old client
+    if 'data' in event:
+        raise ValueError('Upgrade your DLHub SDK to a newer version: pip install -U dlhub_sdk')
+
+    start = time.time()
+    global shim
+    if "shim" not in globals():
+        from home_run import create_servable
+        with open("dlhub.json") as fp:
+            shim = create_servable(json.load(fp))
+    x = shim.run(event["inputs"],
+                 debug=event.get("debug", False),
+                 parameters=event.get("parameters", None))
+    end = time.time()
+    return (x, (end-start) * 1000)
+
+
+def convert_dict(data, conversion_function=str):
```

Comment:
some clarity on what's happening here, please

Reply:
is there any easier/simpler way to approach this? the logic is tough to follow for the reader

Reply:
Some commentary about this being the function registered with funcx and executed within the container would perhaps help clarify this.

Reply:
If you could still add clarity here @ericblau , I'd much appreciate it!